src/api.ts

```diff
@@ -0,0 +1,63 @@
+interface Project {
+  projectName: string;
+  techStack: string;
+  isCompleted: boolean;
+}
+
+interface Developer {
+  id: number;
+  name: string;
+  age: number;
+  isEmployed: boolean;
+  skills: Array<string>;
+  projects: Array<Project>;
+  experience: number;
+}
+
+const developers: Developer[] = [];
+
+function addDeveloper(developer: Developer): void {
+  // validating developer based on type and empty data
+  if (typeof developer.id !== 'number') {
+    throw Error("id is not correct");
+
+  } else if (typeof developer.name !== 'string' || developer.name === '') {
+    // type of name is not string or it's value is empty
+    // it's an error
+    throw Error("Error: name property is either empty or it's type is not correct");
+
+  } else if (typeof developer.age !== 'number' || developer.age < 14 /* boys who are under 14 are not developers */) {
```

Comment:
do not put comments under any code block

either put on top of statement or after the code block - `... developer.age < 14 ) { /**/`